Fix example generator

sgraband opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md
-->

Contributed on behalf of STMicroelectronics.

#### What it does
<!-- Include relevant issues and describe how they are addressed. -->

#### How to test
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->

planger reviewed (commented):
Thank you very much! That looks great! I noticed a few things:
1. We can now delete the `cmake-with-library` example as it is actually equivalent to the one you added
2. In the readme of the `clangd-contexts` example, we should remove the `How to Open the Workspace` section and add the [section describing the usage of the VS Code extension](https://github.com/eclipse-cdt-cloud/clangd-contexts/blob/main/examples/clangd-contexts-ext/README.md#using-the-example-vs-code-extension).
3. We should add a readme that describes what users should do with the `example-traces`, just a few steps on how to open the trace should be sufficient.
4. We should add a plain example project that also generates build tasks (tasks.json) and a launch.json for debugging. This should actually be the first example in the list and also include a short readme what to try (building, opening code, trying code completion, etc, and debugging). I think you had such an example, right?

Thanks a lot!

sgraband commented:
Thanks for the feedback!
I changed the `cmake-with-library` example to a simpler `cmake-example` which can be build and debugged.
I also updated the `clangd-context` readme, but i am not quite sure what i can remove there. Maybe you could take a look?
Lastly, i added the readme for the `example-traces` and renamed the readmes to make them interoperatible so that you can create more than one workspace without the generation breaking.

planger reviewed (approved):
Thank you very much! Looks good to me!
